### PR TITLE
Update CI linter config

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Align CI linter config with scikit-learn-intelex repo.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,12 +16,12 @@
 
 repos:
   - repo: https://github.com/psf/black
-    rev: 23.7.0
+    rev: 24.1.1
```
Comment (david-cortes-intel) [inline .pre-commit-config.yaml:19]: Any particular reason for this version? Latest is 25.1:
https://pypi.org/project/black/#history

Same for `isort` which is at 6.0.1.

**Review by david-cortes-intel — APPROVED** [review]
LGTM, but we might as well set the linters to their latest versions if we're changing them here.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Alexsandruss) [reply .pre-commit-config.yaml:19]: Just copied from sklearnex repo.
